app/forms/date_form.py

```diff
@@ -84,3 +98,62 @@ class MonthYearDateForm(Form):
     MonthYearDateForm.month = SelectField(choices=month_choices, default='', validators=validate_with)
 
     return MonthYearDateForm
+
+
+def get_min_max_dates_for_single_date_period_validation(answer, answer_store, metadata):
+    """
+    Gets attributes within a minimum or maximum of a date field and validates that the entered date
+    is valid.
+
+    :param answer: The answer which contains the minimum or maximum
+    :param answer_store: The current answer store
+    :param metadata: metadata for reference meta dates
+    :return: attributes
+    """
+    minimum_referenced_date, maximum_referenced_date = None, None
+
+    if 'minimum' in answer:
+        minimum_referenced_date = get_referenced_offset_value(answer['minimum'], answer_store, metadata)
+    if 'maximum' in answer:
+        maximum_referenced_date = get_referenced_offset_value(answer['maximum'], answer_store, metadata)
+
+    if minimum_referenced_date and maximum_referenced_date:
```

Comment:
Could you please add a little comment to mention that this should never happen as validator should prevent it.